### PR TITLE
feat(ProfileInfo): JWT token expiration detection

### PR DESCRIPTION
**Pull request by traeok**
**What It Does**

Fernando mentioned that extenders would benefit from having access to the JWT token expiration feature from Zowe Explorer, so I figured it made the most sense to put it the logic for it in `ProfileInfo`. That way, anyone using Imperative to access/manage profiles can leverage this functionality.

**How to Test**

Once merged, this can be tested in action after this PR is updated: https://github.com/zowe/zowe-explorer-vscode/pull/3174

Here's a manual test to show the behavior of the function itself:
- Checkout this branch and run `npm run build && npm pack` from the `packages/imperative` folder
- Copy the `zowe-imperative-8.1.0.tgz` file from this folder and place it into a new folder outside of the repo
- `cd` to the empty folder and run `zowe config init --no-prompt && npm init -y && npm install zowe-imperative-8.1.0.tgz @zowe/secrets-for-zowe-sdk` (you don't need secrets SDK, but its easier than getting around cred manager setup in ProfileInfo)
- Copy this config and replace the contents of `zowe.config.json`:

```jsonc
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "zosmf": {
            "type": "zosmf",
            "properties": {
                "port": 1337
            }
        },
        "project_base": {
            "type": "base",
            "properties": {
                "host": "localhost",
                // this token expired in 2001
                // if you want to test an unexpired token, this one expires in 2128: 123.eyJleHAiOjUwMDAwMDAwMDB9.456
                "tokenValue": "123.eyJleHAiOjEwMDAwMDAwMDB9.456"
            }
        }
    },
    "defaults": {
        "zosmf": "zosmf",
        "base": "project_base"
    }
}
```

- Save the following code snippet as `script.mjs` within this test folder and run it using `node script.mjs`

```javascript
const { ProfileInfo } = await import("@zowe/imperative");
const profInfo = new ProfileInfo("zowe");
await profInfo.readProfilesFromDisk({ projectDir: process.cwd() });
console.log("token expired:", profInfo.hasTokenExpiredForProfile("zosmf"));
```

The script should output `token expired: true`.

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Comment by traeok** [comment]
I don't have permissions on Zowe CLI to update the new issue reported by SonarCloud, but it is a false positive as `argValue` will almost always be type `string`, and even in cases where it isn't a string, all exceptions will be handled in the try/catch block.

**Review by awharn — APPROVED** [review]

**Review by gejohnston — APPROVED** [review]

File: packages/imperative/src/config/src/ProfileInfo.ts [hunk 1]
```diff
@@ -179,6 +179,48 @@ export class ProfileInfo {
...
+                // The Date constructor expects a timestamp in milliseconds.
+                const msPerSec = 1000;
+                const expireDate = new Date(payloadJson.exp * msPerSec);
+                return expireDate < new Date();
+            }
+        } catch (err) {
+            return false;
+        }
+
+        return false;
```
Comment (t1m0thyj) [inline packages/imperative/src/config/src/ProfileInfo.ts:221]: Thanks for adding this to the ProfileInfo API! As the ProfileInfo API is used only by extenders like ZE, I'm curious if we'd want to move this part of the method to make it usable by CLI itself?

Perhaps could have a static `isTokenExpired` method on the `ConfigUtils` or `RestStandaloneUtils` classes, that only checks if the token is expired and doesn't handle loading it from config.

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

File: packages/imperative/src/config/src/ProfileInfo.ts (hunk 1 above)
Comment (traeok) [reply packages/imperative/src/config/src/ProfileInfo.ts]: Good call - I added a function to `ConfigUtils` called `hasTokenExpired` and updated the `ProfileInfo` function to use it.
Also added an entry for the new function in the changelog 👍

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 

looking at the code reminded me of this issue:
- https://github.com/zowe/zowe-cli/issues/1966#issuecomment-1822748769

File: packages/imperative/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+## Recent Changes
+
+- Enhancement: Added a new SDK method (`ConfigUtils.hasTokenExpired`) that allows developers to verify whether a given JSON web token has expired. [#2298](https://github.com/zowe/zowe-cli/pull/2298)
```
Comment (anaxceron) [inline packages/imperative/CHANGELOG.md:7]: I think it'd be better to turn this into an action statement: Use the new SDK method `bbb` to check whether a given JSON web token has expired.
A note: Do you really use the token to verify expiration, as you originally wrote, or to check whether the token has expired?

File: packages/imperative/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
...
+- Enhancement: Added a new SDK method (`ProfileInfo.hasTokenExpiredForProfile`) that allows developers to verify whether a JSON web token has expired for a given profile. [#2298](https://github.com/zowe/zowe-cli/pull/2298)
```
Comment (anaxceron) [inline packages/imperative/CHANGELOG.md:8]: As the previous comment, better to turn this into an action: Use the new SDK method `ccc` to check whether a given JSON web token has expired for a given/specified profile.

**Review by anaxceron — CHANGES_REQUESTED** [review]
Requesting some edits to the changelog

**Review by anaxceron — APPROVED** [review]
Fantastic changelog @traeok!

**Pull request merged**